engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java

```diff
@@ -269,7 +263,16 @@ private void replayUntilDone(final long position, final TypedRecord<?> currentEv
                 zeebeDbTransaction.rollback();
               }
               zeebeDbTransaction = transactionContext.getCurrentTransaction();
-              zeebeDbTransaction.run(operation);
+              zeebeDbTransaction.run(
+                  () -> {
+                    final boolean isNotOnBlacklist =
+                        !zeebeState.getBlackListState().isOnBlacklist(typedEvent);
```

Comment:
The following comment is still valid. We should keep it until it is solved.

```
            // TODO (saig0): ignore blacklist because we replay events only (#7430)
            // these events were applied already
```

Reply:
Sorry I think this got deleted due to resolving merge conflict. But not sure why is this actually necessary? We now replay only right? So this would not be necessary, since we not creating new events on processing?

Reply:
Exactly. We can remove the check against the blacklist. I thought we could remove it as part of #7430.

Reply:
ok :+1: